src/realm/sync/noinst/protocol_codec.hpp

```diff
@@ -254,7 +256,29 @@ class ClientProtocol {
 
                 auto message = msg.read_sized_data<StringData>(message_size);
 
-                connection.receive_error_message(error_code, message, try_again, session_ident); // Throws
+                connection.receive_error_message(error_code, sync::ProtocolErrorInfo{message, try_again},
+                                                 session_ident); // Throws
+            }
+            else if (message_type == "json_error") { // introduced in protocol 4
+                sync::ProtocolErrorInfo info{};
+                int error_code = msg.read_next<int>();
+                auto message_size = msg.read_next<size_t>();
+                auto session_ident = msg.read_next<session_ident_type>('\n');
+                auto json_raw = msg.read_sized_data<std::string_view>(message_size);
+                try {
+                    auto json = nlohmann::json::parse(json_raw);
+                    info.client_reset_recovery_is_disabled = json["isRecoveryModeDisabled"];
+                    info.try_again = json["tryAgain"];
+                    info.message = util::format("%1 Logs: %2", json["message"], json["logURL"]);
```

Comment:
do we want to hide this away inside the protocol codec? it might be worth exposing this out of the sync client at least, and maybe all the way out to the sdks, but provide a `to_string()` method that combines them for easy logging/reporting purposes?

Reply:
In the previous format, the server appended the logURL to the message. I initially didn't do anything extra here because I wanted to just get the server changes in without breaking anything for the SDKs. But I think I found a decent way to break this out in `SyncError` without making any breaking changes for the SDKs.